content/2022/robot-valjak.md

```diff
@@ -2,3 +2,76 @@
+$$L = K_1 + K_2 + T_1 + T_2 - U_1 - U_2$$
+Primenivši Ojler-Lagranžovu jednačinu na dva sistema, klatno i valjak,
+$$\frac{d}{dt}\left(\frac{\partial L}{\partial \omega _ 1}\right)\ -\ \frac{\partial L}{\partial \omega _ 1}\ = \ -T + T_f$$
+$$\frac{d}{dt}\left(\frac{\partial L}{\partial \omega _ 2}\right)\ -\ \frac{\partial L}{\partial \omega _ 2}\ = \ T$$
+dobijaju se sledeće jednačine kojima je u potpunosti opisana dinamika sistema:
+$$-T + T_f\ =\ \alpha_1(J_1 + J_2 + M_1r^2 + M_2r^2 + M_2d^2 - 2M_2rd)\ +\ \alpha_2(J_2 - M_2rd + M_2d^2)\ +\ M_2gd(\theta_1 + \theta_2)$$
+$$T\ =\ \alpha_1(J_1 - M_2rd + M_2d^) + \alpha_2(J_2 + M2d^2) + M_2gd(\theta_1 + \theta_2)$$
+gde su iskorišćene aproksimacije za male uglove i male ugaone brzine.
+
+#### Simulacije
+
+Simulacije služe da se na što realističniji način predstavi realni svet i dobije bolji uvid u to kako će se sistem ponašati. Pored osećaja za sistem, simulacije omogućavaju lakše testiranje softvera kao i testiranje sistema u različitim prirodnim uslovima.
+
+##### Transfer funkcije
+
+Transfer funkcija predstavlja odnos između signala upravljačkog sistema i ulaznog signala, za sve moguće ulazne vrednosti. Za bilo koji kontrolni sistem postoji referentni ulaz ili pobuda koji prolazi kroz transfer funkciju kako bi se proizveo odgovor sistema na odgovarajuću pobudu.
+
+Transfer funkcija (funkcija prenosa) je definisana kao Laplasova transformacija izlazne promenjljive i Laplasove transformacije ulazne promenjljive pod pretpostavkom da su svi početni uslovi jednaki nuli.
+$$G(s)\ =\ \frac{C(s)}{R(s)}$$
+Način na koji se formira transfer funkcija kontrolnog sistema je sledeća:
+
+1. Formiraju se jednačine sistema
+2. Uzimaju se Laplasove transformacije jednačina, pretpostavljajući da su početni uslovi jednaki nuli
+3. Određuju se ulazi i izlazi sistema
+4. Na kraju se uzima odnos Laplasove transformacije izlaza i ulaza što je tražena transfer funkcija
+
+Takođe, važno je naglasiti da izlaz i ulaz nekog sistema ne mora nužno da bude istih fizičkih dimenzija.
+
+U radu su korišćene tri transfer funkcije koje se uz data pravila mogu izvesti iz fizičkih jednačina za horizontalno kretanje valjka. To su transfer funkcija ugla inklinacije klatna valjka i napona na motoru, transfer funkcija ugaone brzine klatna valjka i napona na motoru i, na kraju, ugaona brzina valjka i napona motora.
+
+TODO: Analiza stabilnosti transfer funkcija
+
+##### PID simulacije
+
+Kako bi se razumeo PID kontroler potrebno je definisati odredjene termine. *Željena vrednost* je obično vrednost koju unosi korisnik. Na primer, za sistem za grejanje to bi bila željena temperatura prostorije. *Izlaz* je kontrolisana vrednost PID kontrolera. U automobilu bi to bila količina goriva koja je potrebna da bi se održavala konstantna brzina. *Greška* je vrednost koju koristi PID kontroler da bi odredio kako da manipuliše izlazom kako bi se dostigla željena vrednost. Greška se računa kao razlika željene vrednosti i trenutne vrednosti sistema.
```

Comment:
1. Promenite uvod ovoga. Zašto bi neko hteo da razume PID kontroler?
1. Možda ove stvari poput "ulaz" i "izlaz" da objasnite na početku dinamike, kada objašnjavate transfer funkcije.
1. Ovo za gorivo i automobil mi nije baš najjasnije
1. Bilo bi super ako ubacite šemu kontrolera